### ISSUE TITLE
`rpmautospec`-plugin's git repo check is too strict

### ISSUE THREAD

**Issue by dennisklein** [rpm-software-management/mock#1290]
https://github.com/rpm-software-management/mock/blob/4b3fb22e2f74a17bf7879395b6f6418192430163/mock/py/mockbuild/plugins/rpmautospec.py#L61-L62

This check fails for [git submodules](https://git-scm.com/docs/git-submodule) and [multiple git worktrees](https://git-scm.com/docs/git-worktree). Since [`rpmautospec` already depends on `pygit2`](https://github.com/fedora-infra/rpmautospec/blob/c4086b9d22ec83cb76e06247cc37764152eeda36/pyproject.toml#L47) I propose to improve the git repo check with [`pygit2.discover_repository()`](https://www.pygit2.org/repository.html#pygit2.discover_repository) which already supports submodules and worktrees.

Test with `python -c "import pygit2; import os; print(pygit2.discover_repository(os.getcwd()))"`

Possible implementation:
```python
if not pygit2.discover_repository(host_chroot_sources):
```

**Comment by praiskup** [comment]
@sgallagher fyi

**Comment by praiskup** [comment]
@dennisklein thank you for reporting this!

**Comment by nphilipp** [comment]
@dennisklein good catch! In fact, we don’t want to have `pygit2` as a dependency in this plugin – we changed it intentionally so that the bulk of the code is executed in the build root and outside has only minimal dependencies.

**Comment by nphilipp** [comment]
@praiskup I lean towards implementing this check in `rpmautospec_core` and adapting the mock plugin to use it. What do you think?

**Comment by nphilipp** [comment]
@dennisklein Your check is too lenient, though 😉 (but the error message in the plugin is misleading): it works anywhere in the worktree, but it should only continue there if at the root. I guess I’ll just check for the presence of `.git` and if it’s a plain file, check that it begins with `gitdir:`.

**Comment by dennisklein** [comment]
> but it should only continue there if at the root

true, I didn't realize this aspect.

> I guess I’ll just check for the presence of `.git` and if it’s a plain file, check that it begins with `gitdir: `

:+1: Thank you!

**Comment by praiskup** [comment]
> @praiskup I lean towards implementing this check in rpmautospec_core and adapting the mock plugin to use it. What do you think?

Not sure if the mock plugin needs an update?  But sounds good.

**Comment by nphilipp** [comment]
I’ve looked into it more closely and it seems like just checking these couple files or directories doesn’t cover all the options how a git work tree can be defined. So to me the question is if we can live with not supporting these corner cases – which we could by delegating this step to `git` itself but then it would have to become a dependency (and I’m not sure if it’s worthwhile).

**Comment by dennisklein** [comment]
> So to me the question is if we can live with not supporting these corner cases

To my current use case (submodules) it would be acceptable.

> which we could by delegating this step to `git` itself

This sounds more future-proof as we would not be relying on implementation details of git.
